pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -719,6 +722,24 @@ func (nc *Controller) monitorNodeHealth() error {
 			}
 			if observedReadyCondition.Status == v1.ConditionUnknown {
 				if nc.useTaintBasedEvictions {
+					// If useTaintBasedEvictions is true, we would not evict pods
+					// But we still want to mark Pod.Status.Reason as "NodeLost"
+					selector := fields.OneTermEqualSelector(core.PodHostField, node.Name).String()
+					options := metav1.ListOptions{FieldSelector: selector}
+					pods, err := nc.kubeClient.CoreV1().Pods(metav1.NamespaceAll).List(options)
+					if err != nil {
+						return err
+					}
+					for _, pod := range pods.Items {
+						// Defensive check
+						if pod.Spec.NodeName != node.Name {
+							continue
+						}
+						if _, err = nodeutil.SetPodTerminationReason(nc.kubeClient, &pod, node.Name); err != nil {
+							fmt.Errorf("update status failed for pod %q: %v", format.Pod(&pod), err)
```

Comment:
will the node lost reason in pod be cleared correctly if the node is ready ?

Reply:
@krmayankk In fact, the pod would be deleted as long as node was lost. But since the node was lost. the delete action will not be confirmed until it is ready again. So the flow is like:
1. Node lost
2. node_lifecycle_controller taint the lost node&&  mark pods NodeLost
3. node_lifecycle_controller detect node tainted, kill pods. But this action would stuck since node was lost,  the kill could not be confirmed.
4. Waiting node ready, keep pod in NodeLost
5. Node ready again, pod deletion completed.